src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.java

```diff
@@ -249,32 +265,58 @@ private void manageExcludeFilePattern()
       if (!gitStatus_ ||
           !session_.getSessionInfo().isVcsAvailable(VCSConstants.GIT_ID))
       {
-         ((Element) listPresetExcludeFilePatterns_.getElement().getChild(1))
+         ((Element) listPresetExcludeFilePatterns_.getElement().getChild(
+               Exclude.StandardGit.ordinal()))
             .setAttribute("disabled", "disabled");
          if (listPresetExcludeFilePatterns_.getSelectedIndex() ==
-             ExcludeFilePatterns.StandardGit.ordinal())
-            listPresetExcludeFilePatterns_.setSelectedIndex(ExcludeFilePatterns.None.ordinal());
+             Exclude.StandardGit.ordinal())
+            listPresetExcludeFilePatterns_.setSelectedIndex(Exclude.None.ordinal());
       }
       else
-         ((Element) listPresetExcludeFilePatterns_.getElement().getChild(1))
-            .removeAttribute("disabled");
-
+         ((Element) listPresetExcludeFilePatterns_.getElement().getChild(
+            Exclude.StandardGit.ordinal())).removeAttribute("disabled");
 
       // disable custom filter text box when 'Custom Filter' is not selected
       divExcludeCustomFilter_.getStyle().setDisplay(
             listPresetExcludeFilePatterns_.getSelectedIndex() == 
-            ExcludeFilePatterns.CustomFilter.ordinal()
+            Exclude.CustomFilter.ordinal()
             ? Style.Display.BLOCK
             : Style.Display.NONE);
 
       // user cannot specify include patterns when using git grep
       if (listPresetExcludeFilePatterns_.getSelectedIndex() != 
-          ExcludeFilePatterns.StandardGit.ordinal())
-         listPresetFilePatterns_.setEnabled(true);
+          Exclude.StandardGit.ordinal())
+      {
+         ((Element) listPresetFilePatterns_.getElement().getChild(
+            Include.CommonRSourceFiles.ordinal())).removeAttribute("disabled");
+         ((Element) listPresetFilePatterns_.getElement().getChild(
+            Include.RScripts.ordinal())).removeAttribute("disabled");
+         ((Element) listPresetFilePatterns_.getElement().getChild(
+            Include.CustomFilter.ordinal())).removeAttribute("disabled");
+      }
       else
       {
-         listPresetFilePatterns_.setEnabled(false);
-         listPresetFilePatterns_.setSelectedIndex(IncludeFilePatterns.AllFiles.ordinal());
+         // when using standard git exclusions we don't have the option to search recursively and
+         // specify file types
+         ((Element) listPresetFilePatterns_.getElement().getChild(
```

Comment:
Here and elsewhere: this is verbose and somewhat difficult to read. How about adding something like `DomUtils.setOptionDisabled(element, ordinal, disabled)`?

Reply:
done!